src/Infrastructure.EntityFramework/Vault/Repositories/Queries/CipherDetailsQuery.cs

```diff
@@ -34,7 +34,9 @@ public virtual IQueryable<CipherDetails> Run(DatabaseContext dbContext)
                         FolderId = (_ignoreFolders || !_userId.HasValue || c.Folders == null || !c.Folders.ToLowerInvariant().Contains(_userId.Value.ToString())) ?
                             null :
                             CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, Guid>>(c.Folders)[_userId.Value],
-                        ArchivedDate = c.ArchivedDate,
+                        ArchivedDate = !_userId.HasValue || c.Archives == null || !c.Archives.ToLowerInvariant().Contains(_userId.Value.ToString()) ?
+                            null :
+                            CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, DateTime>>(c.Archives)[_userId.Value],
```

Comment:
❓ I think we may want to be a bit more defense and use a `Dictionary<Guid, DateTime?>` with a nullable `DateTime` in case the DateTime is empty/null.

It shouldn't be possible for users to set an invalid value, but we've seen unexpected values sneak in before so it could be beneficial to be more defensive.